src/Passes/Source/Rules/Notation/Notation.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Rules/Notation/Call.ipp"
+#include "Rules/Operands/Any.hpp"
+#include "Rules/Operands/Call.hpp"
+#include "Rules/Operands/Instruction.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+    namespace notation
+    {
+
+        using IOperandPrototypePtr = std::shared_ptr<IOperandPrototype>;
+
+        class Capture
+        {
+          public:
+            explicit Capture(std::string const& name);
+            // Note that this operator is delibrately unconventional
```

Comment:
Ok - it would be nice to have a bit more insight the deliberation to the comment; what was the reason for that?

Reply:
The reason is that we want to allow notation like:
```c++
"foo"_cap = call("bar")
```
The structure `"foo"_cap` creates a `Capture` object which copies the pointer on the right-handside and returns a R-value that has a specified capture. Since it is an R-value we cannot return a reference as you would do under normal circumstances and the comment is to prevent anybody from removing the "// NOLINT" comment in the following line and return a reference as this would result in a incorrect implementation that would segfault in many cases.

Reply:
I think in that case this should be made more explicit in the comment. 
Something like "do not remove the //NOLINT since returning a reference will result in an incorrect implementation".